src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -629,7 +629,7 @@ impl Step for Miri {
             SourceType::InTree,
             &[],
         );
-        let _guard = builder.msg_sysroot_tool(Kind::Test, compiler.stage, "miri", host, host);
+        let _guard = builder.msg_sysroot_tool(Kind::Test, compiler.stage, "miri", host, target);
```

Comment:
Looks like https://github.com/rust-lang/rust/commit/fff8223584d5c55bfc974772db1856db90e878d8 duplicated a bunch of logic and some of it is wrong? I didn't think each test step has to do its own printing, and I think Miri had a "Testing Miri" header before fff8223584d5c55bfc974772db1856db90e878d8... but I might misremember.

Reply:
> Looks like https://github.com/rust-lang/rust/commit/fff8223584d5c55bfc974772db1856db90e878d8 duplicated a bunch of logic and some of it is wrong?

Others don't support the target field yet.